src/main/java/io/prometheus/cloudwatch/CloudWatchCollector.java

```diff
@@ -83,7 +84,15 @@ private CloudWatchCollector(Map<String, Object> config, AmazonCloudWatchClient c
         }
 
         if (client == null) {
-          this.client = new AmazonCloudWatchClient();
+          if (config.containsKey("role_arn")) {
+            STSAssumeRoleSessionCredentialsProvider credentialsProvider = new STSAssumeRoleSessionCredentialsProvider(
```

Comment:
That is handled by the credentials provider as can be seen [here](https://github.com/aws/aws-sdk-java/blob/master/aws-java-sdk-sts/src/main/java/com/amazonaws/auth/STSAssumeRoleSessionCredentialsProvider.java#L271).